src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -370,10 +363,17 @@ Transactor::checkSingleSign (PreclaimContext const& ctx)
 
     // Consistency: Check signature
     // Verify the transaction's signing public key is authorized for signing.
-    auto const pk =
-        RippleAddress::createAccountPublic(
-            ctx.tx.getSigningPubKey());
-    auto const pkAccount = calcAccountID(pk);
+    auto const spk = ctx.tx.getSigningPubKey();
+    if (!publicKeyType (makeSlice (spk)))
+    {
+        JLOG(ctx.j.trace) <<
+            "checkSingleSign: signing public key type is unknown";
+        return tefBAD_AUTH; // FIXME: should be better error!
```

Comment:
This check needs to be done somewhere else. It's too late for these kinds of checks (checks that permit a transaction to not claim a fee regardless of what's in the ledger) here.